test/spec/services/paymentServicesSpec.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+import i18n from '../../common/script/i18n';
+
+describe('paymentServices', function(){
+  let $httpBackend, $window, user, Payments, getAnySubPattern
+
+  getAnySubPattern= /^\/paypal\/subscribe.*/
+ 
+  beforeEach(module('habitrpg'));
+ 
+  beforeEach(function(){
+    module(function($provide){
+      $window = {href: '', alert: sandbox.spy(), location: {search:   '', pathname: ''}};
+      $provide.value('$window', $window);
+    });
+    
+    inject(function($injector) {
+    // Set up the mock http service responses
+    $httpBackend = $injector.get('$httpBackend');
+    Payments = $injector.get('Payments');
+    
+   // console.log("PAYMENT SERVICES")
```

Comment:
Let's see.  

The regex is just to catch any PayPal sub requests.

beforeEach(module('habitrpg')) I copied from userServicesSpec and may not be necessary.  I didn't think to check, sorry!

The next part stubs out $window, so we can see if Angular triggers an alert when things go wrong or redirects the user when they go right.

Finally, the injector gives us access to a mock $http service and the Payments service we want to test.